### PR TITLE
Feature time and date range filters

### PR DESCRIPTION
**Pull request by bibekdahal**

### CODE REVIEW

**Comment by bibekdahal** [comment]
Used in:

https://github.com/the-deep/client/pull/238

File: components/General/Faram/computer.js [hunk 1]
```diff
@@ -19,7 +20,15 @@ const computeOutputSettings = (obj, schema, dataList = []) => {
...
 
             if (isComputer) {
-                subSettings = { $set: childSchema(...newDataList) };
+                // Update the computed value only if it is different than existing value.
+                // This is to prevent immutability-helper from creating completely
+                // new object even when computed value is not changed.
+                const newData = childSchema(...newDataList);
+                const oldData = (obj || emptyObject)[fieldName];
+
+                subSettings = isEqual(oldData, newData)
```
Comment (tnagorra) [inline components/General/Faram/computer.js:29]: Are we only checking the reference for objects here? Is that the intended behavior?

File: components/General/WrapWithFaram.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                ...otherProps
+            } = this.props;
+
+            const {
+                faramValues,
+                faramErrors,
+            } = this.state;
+
+            const newProps = { ...otherProps };
+            if (submitAction) {
```
Comment (tnagorra) [inline components/General/WrapWithFaram.js:72]: is submitAction a string identifier?
do we really need to inject this.handleSubmit?

File: components/Input/DateFilter/index.js [hunk 3]
```diff
@@ -1,54 +1,107 @@
...
 const defaultProps = {
     className: '',
-    placeholder: 'Select an option',
-    value: undefined,
-    onChange: undefined,
+    value: {},
+    onChange: noOp,
+};
+
+const formatDate = date => FormattedDate.format(decodeDate(date), 'dd-MM-yyyy');
```
Comment (tnagorra) [inline components/Input/DateFilter/index.js]: OMG, FormattedDate.format is not there anymore. 404

File: components/General/Faram/computer.js (hunk 1 above)
Comment (bibekdahal) [reply components/General/Faram/computer.js:29]: yes, that's exactly what we need here

File: components/General/WrapWithFaram.js (hunk 2 above)
Comment (bibekdahal) [reply components/General/WrapWithFaram.js:72]: Yes

Comment (bibekdahal) [reply components/General/WrapWithFaram.js:72]: Consider that we are wrapping a Button with this HOC. Then `submitAction` would be `onClick`.

File: components/Input/DateFilter/index.js (hunk 3 above)
Comment (bibekdahal) [reply components/Input/DateFilter/index.js]: OMG, who moved this without notifying others or keeping backward compatibility. This is 410: resource intentionally is gone.

Comment (tnagorra) [reply components/Input/DateFilter/index.js]: Haha, I did before going to the hackathon.
It was a very bad decision to import format method from FormattedDate.
I thought we had a discussion about moving it

**Pull request merged**